### PR TITLE
Make sure to unregister the producer when timing out a HTTP connection.

### PR DESCRIPTION
**Pull request by Lukasa**
Resolves [Twisted issue 8992](https://twistedmatrix.com/trac/ticket/8992).

### CODE REVIEW

**Comment by Lukasa** [comment]
There is *no way* this patch reduces coverage that much. No chance.

File: src/twisted/web/topfiles/8992.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+twisted.web's HTTP server no longer leaks file descriptors when timing out inactive clients using TLS connections.
```
Comment (exarkun) [inline src/twisted/web/topfiles/8992.bugfix]: Style guide says the subject should be a FQPN.  So perhaps "twisted.web.http.HTTPChannel now ...".  Also, given the comment about how the code is still using `loseConnection` and not `abortConnection`, maybe it's better to claim it tries to fix one case and not imply that connections never leak at all with the new version.

**Review by exarkun — APPROVED** [review]
Thanks.  Looks good.  Just a quibble with the news fragment.  Please address and merge.  Please also file a follow-up ticket for disconnecting non-cooperating clients (ie, using abortConnection).  Which maybe belongs in TimeoutMixin?  Or as part of an even more general "try to close gracefully but after a while just kill it" API.

**Comment by exarkun** [comment]
Also, per development guidelines, please file a ticket for the unrelated failure at https://buildbot.twistedmatrix.com/builders/windows7-64-py2.7-coverage/builds/711/steps/default/logs/problems :/

**Comment by Lukasa** [comment]
Thanks @exarkun. The follow-up ticket was in fact already opened by past-me as [8902](https://twistedmatrix.com/trac/ticket/8902). Tracking ticket for the build failure has just been opened as [8994](https://twistedmatrix.com/trac/ticket/8994). Topfile has been reworded.

I'll let the builders run again and then merge. Thanks for the review!

**Pull request merged**